From 76059ee8362f867a9258023190dd8865d72688af Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 11 Aug 2017 18:44:03 +0800
Subject: [PATCH 2377/5242] MLK-16197-2 video: fbdev: dcss: init dcss irqs
 earlier

commit  19543f1a0b4af2601897455ccbd612cf2901e18a from
https://source.codeaurora.org/external/imx/linux-imx.git

Initialize the DCSS interrupt system earlier to make
it enabled before any cfifo commit.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 72adfb2..7e7e81d 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -3246,6 +3246,8 @@ static int dcss_probe(struct platform_device *pdev)
 	/* Pull DCSS out of resets */
 	writel(0xffffffff, info->blkctl_base + 0x0);
 
+	dcss_interrupts_init(info);
+
 	/* register channel 0: graphic */
 	ret = dcss_register_one_ch(0, info);
 	if (ret)
@@ -3256,8 +3258,6 @@ static int dcss_probe(struct platform_device *pdev)
 	if (ret)
 		goto unregister_ch0;
 
-	dcss_interrupts_init(info);
-
 	/* enable encoder if exists */
 	dcss_enable_encoder(info);
 
-- 
1.7.9.5

